### ISSUE TITLE
Fix warnings in `Vc` that arise when compiling ROOT

### ISSUE THREAD

**Issue by guitargeek** [VcDevel/Vc#355]
When including `Vc.h`, ROOT Math disables several compiler diagnostics to avoid warnings in the CI and nightlies:
https://github.com/root-project/root/blob/0fd27057b7b25e777ea642dfcc216ab27a6ece0d/math/mathcore/inc/Math/Types.h

It would be good if any warnings can be avoided in upstream veccore, such that we don't need this hack in ROOT.

See also https://github.com/root-project/veccore/issues/24.

**Comment by bernhardmgruber** [comment]
Hi! Vc is no longer developed. If you can provide a PR, I can help you get it in. In the long term I would try to migrate to `std::simd`. There is a fallback implementation available if you have to support compilers without `std::simd`: https://github.com/mattkretz/vir-simd

**Comment by dpiparo** [comment]
for the time being: https://github.com/VcDevel/Vc/pull/356 . If you are happy with the pr, your test suite and nightlies pass, could you tag a new version so that it can be included e.g. in ROOT? Thanks a lot in advance!!

**Comment by bernhardmgruber** [comment]
I merged the PR on Vc.

**Comment by bernhardmgruber** [comment]
@guitargeek, does PR #356 entirely solve this issue?

**Comment by guitargeek** [comment]
No, not entirely. It solves the current remaining warning in the nighlies, yes, but we have still "hotfixed" away several warnings in ROOT Math:
https://github.com/guitargeek/root/blob/master/math/mathcore/inc/Math/Types.h#L11

Once Vc is in a state where we get no warnings in the ROOT nightlies without the pragmas, then I would consider this issue as solved.

Thanks for the work you have done already, @bernhardmgruber and @dpiparo :+1:

**Comment by guitargeek** [comment]
I'm closing the issue because ROOT doesn't use `Vc` anymore.
